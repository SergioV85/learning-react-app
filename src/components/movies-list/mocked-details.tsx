export const mockedMovieDetails = {
  "budget": 0,
  "id": 419831,
  "overview": "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
  "poster_path": "https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg",
  "release_date": "2018-01-21",
  "revenue": 0,
  "runtime": 106,
  "tagline": "",
  "title": "I Kill Giants",
  "vote_average": 6,
  "vote_count": 63,
  "genres": [
    "Thriller",
    "Drama",
    "Fantasy"
  ],
}