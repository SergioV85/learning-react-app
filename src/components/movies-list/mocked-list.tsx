export const mockedMoviesList = [{
    "id": 419831,
    "title": "I Kill Giants",
    "tagline": "",
    "vote_average": 6,
    "vote_count": 63,
    "release_date": "2018-01-21",
    "poster_path": "https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg",
    "overview": "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
    "budget": 0,
    "revenue": 0,
    "genres": [
      "Thriller",
      "Drama",
      "Fantasy"
    ],
    "runtime": 106
  },
  {
    "id": 399057,
    "title": "The Killing of a Sacred Deer",
    "tagline": "",
    "vote_average": 6.6,
    "vote_count": 509,
    "release_date": "2017-10-20",
    "poster_path": "https://image.tmdb.org/t/p/w500/e4DGlsc9g0h5AyoyvvAuIRnofN7.jpg",
    "overview": "A teenager's attempts to bring a brilliant surgeon into his dysfunctional family take an unexpected turn.",
    "budget": 0,
    "revenue": 2205074,
    "genres": [
      "Drama",
      "Horror",
      "Thriller"
    ],
    "runtime": 121
  },
  {
    "id": 423988,
    "title": "Kill Switch",
    "tagline": "Worlds will collide",
    "vote_average": 5,
    "vote_count": 81,
    "release_date": "2017-06-01",
    "poster_path": "https://image.tmdb.org/t/p/w500/8K6pfMcT3hiIlCFw2RrHoXLshxl.jpg",
    "overview": "A pilot battles to save his family and the planet after an experiment for unlimited energy goes wrong.",
    "budget": 1000000,
    "revenue": 0,
    "genres": [
      "Science Fiction"
    ],
    "runtime": 91
  },
  {
    "id": 382322,
    "title": "Batman: The Killing Joke",
    "tagline": "The madness begins.",
    "vote_average": 6.2,
    "vote_count": 618,
    "release_date": "2016-07-21",
    "poster_path": "https://image.tmdb.org/t/p/w500/zm0ODjtfJfJW0W269LqsQl5OhJ8.jpg",
    "overview": "As Batman hunts for the escaped Joker, the Clown Prince of Crime attacks the Gordon family to prove a diabolical point mirroring his own fall into madness. Based on the graphic novel by Alan Moore and Brian Bolland.",
    "budget": 3500000,
    "revenue": 3775000,
    "genres": [
      "Action",
      "Animation",
      "Crime",
      "Drama"
    ],
    "runtime": 72
  },
  {
    "id": 237756,
    "title": "Kill Me Three Times",
    "tagline": "Once is never enough.",
    "vote_average": 5.5,
    "vote_count": 203,
    "release_date": "2015-04-10",
    "poster_path": "https://image.tmdb.org/t/p/w500/zqTlUUBlVZTzVMYUnJQve0kJc3o.jpg",
    "overview": "While on a seemingly routine job, a jaded hit man discovers that he's not the only one with his target in the crosshairs.",
    "budget": 0,
    "revenue": 0,
    "genres": [
      "Comedy",
      "Thriller"
    ],
    "runtime": 90
  }
]