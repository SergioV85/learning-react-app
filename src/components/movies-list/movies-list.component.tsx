import * as React from 'react';
import { ErrorBoundary } from './../error-boundary/error-boundary.component';
import { IMovieObject, MovieCard } from './../movie-card/movie-card.component';
interface IMoviesListProps {
  movies: IMovieObject[];
  onMovieClick(): void;
}

export class MoviesList extends React.Component<IMoviesListProps, {}> {
  constructor(props: IMoviesListProps) {
    super(props);
  }

  public render() {
    return <div className={`c-movies-list ${this.props.movies.length === 0 ? 'c-movies-list--empty' : ''}`}>
      {this.listOrMessage}
    </div>;
  }

  private get listOrMessage() {
    return this.props.movies.length === 0
    ? <h1 className='c-movies-list__no-found-title'>No films found</h1>
    : <ul className='c-movies-list__collection container'>
        {this.props.movies.map((movie) => this.getMovieCard(movie))}
      </ul>;
  }

  private getMovieCard(movie: IMovieObject) {
    return <ErrorBoundary
      key={`movie-card-wrapper-${movie.id}`}
      errorMessage={`Unfortunately, we can't show ${movie.title} preview`}
    >
      <MovieCard key={movie.id} movie={movie} onMovieClick={this.props.onMovieClick} />
    </ErrorBoundary>;
  }
}
