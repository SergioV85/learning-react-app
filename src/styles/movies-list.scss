@import './variables';

.c-movies-list {
  display: flex;
  height: 100%;

  &--empty {
    align-items: center;
    justify-content: center;
  }
  &__collection {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding: 0;
    width: 100%;
  }
}