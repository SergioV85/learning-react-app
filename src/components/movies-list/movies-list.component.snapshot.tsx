import * as React from 'react';
import { shallow } from 'enzyme';
import { MoviesList } from './movies-list.component';
import { mockedMoviesList } from './mocked-list';

describe('Movies List Component Snapshot', () => {
  it('should renders correctly', () => {
    const moviesListSnapshot = shallow(
      <MoviesList movies={mockedMoviesList} onMovieClick={() => {}} />
    );
    expect(moviesListSnapshot)
      .toMatchSnapshot();
  });
});