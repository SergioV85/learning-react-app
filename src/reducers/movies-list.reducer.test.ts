// import { MoviesListActions } from './../actions/movies-list.actions';
import { moviesListReducer } from './movies-list.reducer';

const mockedState = {
  isLoading: false,
};

const initialState = moviesListReducer(mockedState, {} as any);

describe('Movies List Reducer', () => {
  describe('initial state', () => {
    it('should match a snapshot', () => {
      expect(initialState).toMatchSnapshot();
    });
  });
});
