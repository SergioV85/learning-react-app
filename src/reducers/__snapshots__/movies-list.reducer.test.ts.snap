// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies List Reducer initial state should match a snapshot 1`] = `
Object {
  "isLoading": false,
}
`;
